### PR TITLE
Update editor only once after execution of save actions

### PR DESCRIPTION
**Pull request by sschaef**
Updating the editor is expensive, therefore we only want to do it a
single time - at the end of the save action running phase.

Fixes #1002309

### CODE REVIEW

File: org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala [hunk 1]
```diff
@@ -102,18 +120,39 @@ trait SaveActionExtensions extends HasLogger {
...
-    undoManager.endCompoundChange()
+      val undoManager = DocumentUndoManagerRegistry.getDocumentUndoManager(udoc)
+      undoManager.beginCompoundChange()
+
+      try applyDocumentExtensions(udoc)
+      catch {
+        // thrown when save actions had to be aborted
+        case e: IllegalStateException =>
+          updateEditor()
+          throw e
```
Comment (dragos) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: Is there no need to call `endCompoundChange()` when an exception is thrown? Or should we call it inside a `finally` block?

Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala]: Indeed, that is a bug. I'll wrap the entire code block in a finally clause.

**Comment by dragos** [comment]
Anyone tried out this PR? I won't have time for code reviews in the next days, so any help is appreciated.

**Comment by dragos** [comment]
LGTM! Thanks @sschaef

**Pull request merged**